### ISSUE TITLE
ACE mortar rounds not in arsenal

### ISSUE THREAD

**Issue by shukari** [acemod/ACE3#5592]
**Arma 3 Version:** stable
**CBA Version:** stable
**ACE3 Version:** stable

**Mods:**
```
- CBA_A3
- ace
```

**Description:**
All this mortar rounds dont show up in the arsenal... so if you add them to a supplybox per mod and then edit the box in eden... the mortar rounds are not shown because the arsenal dont know this rounds...
"ACE_1Rnd_82mm_Mo_HE",
"ACE_1Rnd_82mm_Mo_Smoke",
"ACE_1Rnd_82mm_Mo_Illum",

**Steps to reproduce:**
- get full arsenal
- try endless to get the mortar rounds

**Where did the issue occur?**
- Dedicated / Self-Hosted Multiplayer / Singleplayer / Editor (Singleplayer) / Editor (Multiplayer) / Virtual Arsenal

**Comment by AACO** [comment]
If the magazine config `type` is set to magazine (multiple of 256) it should show up in the arsenal, I'll test it out and try to create a PR for this.

**Comment by kripto202** [comment]
mortar rounds are considered as vehicle ammo

**Comment by jarckn** [comment]
Hey,

whats the status of this issue? It, seems, that the ammo is still not viewable in 3.12.
Tested in arsenal inside a mission and in the virtual arsenal.

Cheers

**Comment by shukari** [comment]
this issue seems to stall

**Comment by dedmen** [comment]
>All this mortar rounds dont show up in the arsenal... so if you add them to a supplybox per mod and then edit the box in eden... the mortar rounds are not shown because the arsenal dont know this rounds

Arsenal doesn't work on boxes.


Is this a ACE Arsenal issue too?

**Comment by shukari** [comment]
Then is is more a feature request, it would be nice to have the mortar rounds in the (ace) arsenal.
ATM the only way to get the mortar rounds is per supplyboxes.

**Comment by dedmen** [comment]
I agree. But quite sure @alganthe already said something about that.

**Comment by alganthe** [comment]
They're type 0, vehicle mags aren't added automatically for obvious reasons.
You can add them manually by doing:
```sqf
["ace_arsenal_displayOpened", {
    (ace_arsenal_virtualItems select 2) append ["ACE_1Rnd_82mm_Mo_HE","ACE_1Rnd_82mm_Mo_Smoke","ACE_1Rnd_82mm_Mo_Illum",
"ACE_1Rnd_82mm_Mo_HE_Guided","ACE_1Rnd_82mm_Mo_HE_LaserGuided"];
}] call CBA_fnc_addEventHandler;
```
or if you don't want it to apply to every box:
```sqf
(_myBox getVariable "ace_arsenal_virtualItems") select 2 append ["ACE_1Rnd_82mm_Mo_HE","ACE_1Rnd_82mm_Mo_Smoke","ACE_1Rnd_82mm_Mo_Illum",
"ACE_1Rnd_82mm_Mo_HE_Guided","ACE_1Rnd_82mm_Mo_HE_LaserGuided"];
```

**Comment by thojkooi** [comment]
@alganthe is there anything preventing us from adding the mortar rounds to ACE arsenal easily?

Going with the manual approach may be an acceptable workaround, if we need to change a lot for this to work.

**Comment by alganthe** [comment]
@thojkooi nothing really, outside the fact that it's a vehicle magazine (for good reasons) meaning it's not auto-picked by ace arsenal.

here's why not filtering vehicles mags isn't a good idea:
![img](https://i.imgur.com/e7FZCwB.jpg)

**Comment by jonpas** [comment]
I'd just allow force adding specific ones like mortar rounds.

**Comment by alganthe** [comment]
^ because the initial array is built at preStart, are all addons loaded at that point?
I mean, I could, not sure if I should.

**Comment by jonpas** [comment]
Not the initial array, I mean add them via the event above, on open.

**Comment by alganthe** [comment]
That would potentially break whitelisting / blacklisting.

**Comment by jonpas** [comment]
`preStart = 1; //1 to call the function upon game start, before title screen, but after all addons are loaded (config.cpp only)`

**Comment by PabstMirror** [comment]
Will be fixed in 3.12.4 (only for ace arsenal)
